components/script/script_thread.rs

```diff
@@ -1526,6 +1535,33 @@ impl ScriptThread {
         document.run_the_animation_frame_callbacks();
     }
 
+    /// Handles firing of transition events.
+    #[allow(unsafe_code)]
+    fn handle_transition_event(&self, unsafe_node: UnsafeNode, name: String, duration: f64) {
+        let node = unsafe { ServoLayoutNode::from_unsafe(&unsafe_node) };
+        let node = unsafe { node.get_jsmanaged().get_for_script() };
+
+        if let Some(el) = node.downcast::<Element>() {
+            if &*window_from_node(node).GetComputedStyle(el, None).Display() == "none" {
+                return;
+            }
+        }
+
+        let init = TransitionEventInit {
+            parent: EventInit {
+                bubbles: true,
+                cancelable: false,
+            },
+            propertyName: DOMString::from(name),
+            elapsedTime: Finite::new(duration as f32).unwrap(),
+            pseudoElement: DOMString::new()
```

Comment:
Do we handle pseudo-elements at all?

It seems to me that we we might be animating the node if the pseudo-element style is animated instead, though that seems a more general issue not related to the DOM bits only (I'm on an airport right now and can't check easily).

If you don't fix it as part of this issue (which would be awesome), please open an issue about it and CC me, then reference it from here.